src/data_readers/data_reader.cpp

```diff
@@ -81,39 +78,35 @@ void generic_data_reader::shuffle_indices(rng_gen& gen)
 void generic_data_reader::setup(int num_io_threads,
                                 observer_ptr<thread_pool> io_thread_pool)
 {
-  m_base_offset = 0;
-  m_sample_stride = 1;
-  m_stride_to_next_mini_batch = 0;
-  m_stride_to_last_mini_batch = 0;
-  m_current_mini_batch_idx = 0;
-  m_num_iterations_per_epoch = 0;
-
-  set_initial_position();
-
   shuffle_indices();
 
   m_io_thread_pool = io_thread_pool;
 }
 
 int lbann::generic_data_reader::fetch(std::vector<conduit::Node>& samples,
                                       El::Matrix<El::Int>& indices_fetched,
-                                      size_t mb_size)
+                                      El::Int current_position_in_data_set,
+                                      El::Int sample_stride,
+                                      size_t mb_size,
+                                      const execution_mode mode)
 {
   // Check to make sure that a valid map was passed
   if (samples.empty()) {
     LBANN_ERROR("fetch function called with no valid buffers");
   }
 
-  if (!position_valid()) {
-    if (position_is_overrun()) {
+  // BVE FIXME
+  if (!(current_position_in_data_set < get_num_data()) /*position_valid()*/) {
+    if (current_position_in_data_set >=
+        get_num_data() /*position_is_overrun()*/) {
       return 0;
     }
     else {
-      LBANN_ERROR(std::string{} +
-                  "generic data reader load error: !position_valid" +
-                  " -- current pos = " + std::to_string(m_current_pos) +
-                  " and there are " +
-                  std::to_string(m_shuffled_indices.size()) + " indices");
+      LBANN_ERROR(
+        std::string{} + "generic data reader load error: !position_valid" +
+        " -- current pos = " + std::to_string(current_position_in_data_set) +
+        " and there are " + std::to_string(m_shuffled_indices.size()) +
+        " indices");
```

Comment:
Since you touched this anyway, it should have been updated to the new syntax:
```
      LBANN_ERROR(
        "generic data reader load error: !position_valid -- current pos = ",
        current_position_in_data_set, " and there are ",
        m_shuffled_indices.size(), " indices");
```
or however `clang-format` would handle that.